# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/ActorsChild.h
# Commit: b525ad6838f0
# Full Hash: b525ad6838f091c2ab7cfc52cda593eca7941fa8
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2019-11-05 22:07:34
# Regressor Bug: 1168606
# File Overlap Count: 1
# Description:
#   Bug 1168606 - Add support for preloading key-only cursors. r=ttung,asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44009
# ==============================================================================

diff -r c009bbdbbfd5 -r b525ad6838f0 dom/indexedDB/ActorsChild.h
--- a/dom/indexedDB/ActorsChild.h	Tue Nov 05 14:40:35 2019 +0000
+++ b/dom/indexedDB/ActorsChild.h	Tue Nov 05 14:40:31 2019 +0000
@@ -633,6 +633,8 @@
     CachedResponse(Key aKey, StructuredCloneReadInfo&& aCloneInfo);
     CachedResponse(Key aKey, Key aLocaleAwareKey, Key aObjectStoreKey,
                    StructuredCloneReadInfo&& aCloneInfo);
+    explicit CachedResponse(Key aKey);
+    CachedResponse(Key aKey, Key aLocaleAwareKey, Key aObjectStoreKey);
 
     CachedResponse(CachedResponse&& aOther) = default;
     CachedResponse& operator=(CachedResponse&& aOther) = default;
@@ -717,7 +719,7 @@
 
   void HandleResponse(const nsTArray<ObjectStoreCursorResponse>& aResponses);
 
-  void HandleResponse(const ObjectStoreKeyCursorResponse& aResponse);
+  void HandleResponse(const nsTArray<ObjectStoreKeyCursorResponse>& aResponses);
 
   void HandleResponse(const nsTArray<IndexCursorResponse>& aResponses);
 
@@ -728,7 +730,7 @@
   void HandleMultipleCursorResponses(const nsTArray<T>& aResponses,
                                      const Func& aHandleRecord);
 
-  void HandleResponse(const IndexKeyCursorResponse& aResponse);
+  void HandleResponse(const nsTArray<IndexKeyCursorResponse>& aResponses);
 
   // IPDL methods are only called by IPDL.
   virtual void ActorDestroy(ActorDestroyReason aWhy) override;